### PR TITLE
Remove some unused variables and properties

### PR DESCRIPTION
**Pull request by tgalopin**
| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT

Analyzing symfony/symfony using https://insight.sensiolabs.com, I found several chunks of dead code. This PR removes them in the 3.3 branch.

### CODE REVIEW

**Comment by lyrixx** [comment]
@tgalopin did you run Insight on lower maintained branch too ?

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/DateIntervalToStringTransformer.php [hunk 1]
```diff
@@ -36,7 +35,6 @@ class DateIntervalToStringTransformer implements DataTransformerInterface
     public function __construct($format = 'P%yY%mM%dDT%hH%iM%sS', $parseSigned = false)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/DateIntervalToStringTransformer.php]: argument can be remved

**Comment by tgalopin** [comment]
@lyrixx yes, I ran it on 2.7 (merged) and 2.8 (no change worthy of being merged).

**Review by vudaltsov — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @tgalopin.

**Pull request merged**